### ISSUE TITLE
Handle Anthropic overloaded errors during streaming

### ISSUE THREAD

**Issue by doomspork** [brainlid/langchain#193]
As we're making more use of streaming we're seeing more of these errors recently:

```elixir
[error] Unsupported event received when parsing Anthropic response: "event: error\ndata: {\"type\":\"error\",\"error\":{\"details\":null,\"type\":\"overloaded_error\",\"message\":\"Overloaded\"}        }"
```

The result is this error logged but the code seems to proceed like everything is okay. I would expect/prefer an error were raised so we could retry the conversation versus leaving it in a partial and incomplete state.

I started spelunking today but it looks like handling this could be a decent sized changes so I thought it was best to bring it here for discussion.

Anthropic documentation around streaming errors for reference: https://docs.anthropic.com/en/api/messages-streaming#error-events

**Comment by brainlid** [comment]
Just saw this myself this morning! Thanks for reporting it.

**Comment by doomspork** [comment]
I haven't had a chance to pull your changes and do my own testing but I will here shortly.

It's unfortunate how frequently their API is overloaded recently.

**Comment by brainlid** [comment]
Yeah. I’m going to take a different approach. I’ll try raising an exception
and catching it in the LLMChain.

So no, it’s not working great yet.

**Comment by brainlid** [comment]
@doomspork I pushed up a hopefully more complete fix. 

When it happens again, the `LLMChain.run(chain, ...)` result will be: `{:error, updated_chain, %LangChain.LangChainError{} = error}`.

Where the error has a `type` of "overloaded_error" (that's what they return).

If the pattern holds, it will be down again tomorrow morning (but it's the weekend, so perhaps not). I'll be watching for it too though.

**Comment by doomspork** [comment]
Sounds good @brainlid! Thanks for tackling this one 🙏  I'll let you know what I see 🫡

**Comment by brainlid** [comment]
Maybe extra fixed now? #500

**Comment by doomspork** [comment]
Awesome work @brainlid! Unfortunately we moved away from using the library with our Elixir application so I'm unable to test currently but continuing to keep an eye on the progress 🎉  Cheers!

**Comment by brainlid** [comment]
No worries, thanks for the update!
